### ISSUE TITLE
EKS version upgrade issue

### ISSUE THREAD

**Issue by winter7** [Altinity/clickhouse-operator#502]
## Situation
I had this plan to upgrade my EKS cluster version, which means I got to execute `drain node` on every existing node.

My clickhouse-operator was on `node A`, My ClickHouse instances were installed on other separate `nodes`.
When I launched a new `node A1` for clickhouse-operator, and ran `drain node A`, suddenly all my ClickHouse instances on other nodes which are not yet upgraded restarted **at the same time**.

This is a critical situation. Can't take this risk in product environment.
## Reproduce
I tried to restore that scene in the develop environment but faild. I deleted clickhouse-operator pod, which I think would have the same effect as upgrading EKS, and found nothing happened. ClickHouse instances are still running.
Does this have to do with different versions of 2 nodes? 

## Version and Expect
My clickhouse-operator version: `0.9.9`.
I expect clickhouse instances not to restart or at least scroll to restart when upgrading EKS version of clickhouse-operator's node. I'm considering migrating clickhouse-operator to one of ClickHouse nodes, but I think this issue will not be affected.

**Comment by winter7** [comment]
I find something like this:
```
      "Update ConfigMap clickhouse/chi-cluster-deploy-confd-clickhouse-replica0",
      "Update StatefulSet(clickhouse/chi-cluster-clickhouse-replica0) - started",
      "Update StatefulSet(clickhouse/chi-cluster-clickhouse-replica0) - failed with error\n---\nStatefulSet.apps \"chi-cluster-clickhouse-replica0\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden\n--\nContinue with recreate",
      "Create StatefulSet clickhouse/chi-cluster-clickhouse-replica0 - started",
      "Create StatefulSet clickhouse/chi-cluster-clickhouse-replica0 - completed",
```
Is it possibly a bug that clickhouse-operator cannot modify properties in a statefuleset?

**Comment by alex-zaitsev** [comment]
@winter7, did you use the same version of ClickHouse operator on the old and new node?

**Comment by winter7** [comment]
@alex-zaitsev, the version of clickhouse-operator on both nodes remains same.
And I believe the first simultaneous reboot I observed was a miscalculation. They were actually rolling reboots.

Here is more information:
- I did modify something in ConfigMap. ( In fact it is `</macros>` in cm `chi-name-deploy-confd-clickhouse-s0-r0`. Not sure if this caused clickhouse-operator to attempt to modify ConfigMap back to the default, and when failing to do so, chi was restarted directly.
- As soon as clickhouse-perator migrates to another node, CHI will restart with Rolling regardless of whether the new node's EKS version has changed. (It seems has nothing to do with EKS version upgrade.)

So the question should be, sorry for inconvenience, why does operator pod migrating to a another node causes clickhouse restart? If it is due to my modification of configmap which was generated by operator, then why does operator fail to modify it back to default?
